nucypher/network/server.py

```diff
@@ -222,13 +225,40 @@ def set_policy(self, id_as_hex, request: Request):
         # TODO: Sign the arrangement here.  #495
         return  # TODO: Return A 200, with whatever policy metadata.
 
+    def revoke_arrangement(self, id_as_hex, request: Request):
+        """
+        REST endpoint for revoking/deleting a KFrag from a node.
+        TODO: Have Ursula return a signed receipt.
+        """
+        from nucypher.policy.models import RevocationNotice
+
+        revocation_notice = RevocationNotice.from_bytes(request.body)
+
+        try:
+            with ThreadedSession(self.db_engine) as session:
+                # Verify the Notice was signed by Alice
+                policy_arrangement = self.datastore.get_policy_arrangement(
+                    revocation_notice.arrangement_id.hex().encode(),
+                    session=session)
+                alice_pubkey = UmbralPublicKey.from_bytes(
+                    policy_arrangement.alice_pubkey_sig.key_data)
+                is_valid = revocation_notice.verify(alice_pubkey)
+
+                if is_valid:
+                    self.datastore.del_policy_arrangement(
+                        revocation_notice.arrangement_id.hex().encode(),
+                        session=session)
+        except:
+            return 404
+        return 200
+
     def reencrypt_via_rest(self, id_as_hex, request: Request):
         from nucypher.policy.models import WorkOrder  # Avoid circular import
         arrangement_id = binascii.unhexlify(id_as_hex)
         work_order = WorkOrder.from_rest_payload(arrangement_id, request.body)
         self.log.info("Work Order from {}, signed {}".format(work_order.bob, work_order.receipt_signature))
         with ThreadedSession(self.db_engine) as session:
-            policy_arrangement = self.datastore.get_policy_arrangement(arrangement_id=id_as_hex.encode(),
+            policy_arrangement = self.datastore.get_policy_arrangement(id_as_hex.encode(),
```

Comment:
:-1:  - Once again confused about types here, and we lost some semantic value, too

Reply:
It doesn't bother me that much, especially since we're `binascci`'ing above.

Reply:
Why do we have it as hex (rather than the pure bytestring) here btw?